pkgs/os-specific/linux/kernel-headers/4.4.nix

```diff
@@ -1,43 +1,35 @@
-{ stdenv, fetchurl, perl, cross ? null }:
+{ stdenvNoCC, lib, buildPackages
+, buildPlatform, hostPlatform
+, fetchurl, perl
+}:
 
-assert cross == null -> stdenv.isLinux;
+assert hostPlatform.isLinux;
 
 let
-
   version = "4.4.10";
-
-  kernelHeadersBaseConfig =
-    if cross == null
-    then stdenv.platform.kernelHeadersBaseConfig
-    else cross.platform.kernelHeadersBaseConfig;
-
+  inherit (hostPlatform.platform) kernelHeadersBaseConfig;
 in
 
-stdenv.mkDerivation {
+stdenvNoCC.mkDerivation {
```

Comment:
Um, why would this use `stdenvNoCC` given that the build needs to compile some stuff?

Reply:
Such thing also belongs to the standard build environment even when cross compiling (e.g Buildroot and Yocto do). If nixpkgs doesn't anymore then *that* needs fixing instead.